### ISSUE TITLE
Should you be able to set mask inplace on a Map?

### ISSUE THREAD

**Issue by Cadair** [sunpy/sunpy#3166]
We strongly discourage almost all other inplace modifications of a Map, why is mask special?

**Comment by wtbarnes** [comment]
Yes. A mask is not metadata though it may be useful at times to derive that mask from metadata. Typically, a mask (unlike metadata) is not specified by a data provider. One may also need to modify a mask depending on the kind of analysis you are doing, e.g. you may want to mask out negative values that resulted from a pedestal subtraction prior to creating a L1 data product if you're doing some sort of thermal analysis on your images.

**Comment by Cadair** [comment]
I guess we allow in-place modification of data (although not with our own methods). So I'm happy to keep the ability to do it but we shouldn't do it in any of our functions or methods?

**Comment by wtbarnes** [comment]
Agreed though I think setting the initial mask during instantiation should be allowed, e.g. our discussion on #8519
